dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -385,6 +385,15 @@ public Workflow updateWorkflow(@ApiParam(hidden = true) @Auth User user,
         c.updateInfo(workflow);
 
         Workflow result = workflowDAO.findById(workflowId);
+
+        //update the workflow path in all workflowVersions
+        Set<WorkflowVersion> versions = c.getVersions();
```

Comment:
Two comments
1) It isn't actually necessary to convert everything from a Set into a List, you can use a set in an enhanced for-loop in Java anyway
2) More importantly, you don't want to set the path of all versions to the result default path all the time. This way, even if we wanted to change something entirely unrelated like the organization, it would still make this change. 

We should probably create a new resource like "/{workflowId}/resetVersionPaths" for example